File: src/types/subscription/subscription.ts
import {z} from "zod";

export const subscriptionPlanSchema = z.enum(["monthly", "yearly"]);

export const subscriptionStatusSchema = z.enum([
  "active", // Aktif odeme yapilmis abonelik
  "expired", // Abonelik suresi doldu
  "cancelled", // Abonelik iptal edildi
]);

export const subscriptionSchema = z.object({
  _id: z.string().optional(),
  companyId: z.string(),
  status: subscriptionStatusSchema,

  // Plan Bilgisi
  plan: subscriptionPlanSchema,

  // Paytr Ödeme Bilgisi
  paytrMerchantOid: z.string(),
  paytrPaymentAmount: z.number(),

  subscriptionStartDate: z.date(),
  subscriptionEndDate: z.date(),

  createdAt: z.date(),
});

export const subscriptionInfoSchema = z.object({
  status: subscriptionStatusSchema,
  plan: subscriptionPlanSchema,
  expiresAt: z.date(),
});

export type Subscription = z.infer<typeof subscriptionSchema>;
export type SubscriptionInfo = z.infer<typeof subscriptionInfoSchema>;
export type SubscriptionPlan = z.infer<typeof subscriptionPlanSchema>;
export type SubscriptionStatus = z.infer<typeof subscriptionStatusSchema>;

File: fisoft/src/main/kotlin/types/subscription/Subscription.kt
package com.ilyasdemirkiran.types.subscription

import com.ilyasdemirkiran.types.Money
import kotlinx.serialization.Serializable
import org.bson.BsonType
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.codecs.pojo.annotations.BsonRepresentation

@Serializable
data class Subscription(
    @BsonId @BsonRepresentation(BsonType.OBJECT_ID) val _id: String? = null,
    @BsonRepresentation(BsonType.OBJECT_ID) val companyId: String? = null,
    val status: String = "active",
    val plan: String = "monthly",
    val paytrMerchantOid: String = "",
    val paytrPaymentAmount: Money = Money.ZERO,
    val subscriptionStartDate: String = "",
    val subscriptionEndDate: String = "",
    val createdAt: String = ""
)

@Serializable
data class SubscriptionInfo(
    val status: String = "",
    val plan: String = "",
    val expiresAt: String = ""
)
